Ark.Tools.ResourceWatcher/IResourceInfo.cs

```diff
@@ -13,7 +14,11 @@ public interface IResourceMetadata
         /// <summary>
         /// The "version" of the resource. Used to avoid retrival of the resource in case if the same version is already been processed successfully
         /// </summary>
-        LocalDateTime Modified { get; }
+        LocalDateTime? Modified { get; }
+        /// <summary>
+        /// The "versions" of the resource. Used to manage multiple sources for the resource to avoid retrival of the resource in case if the last version is already been processed successfully

```

Comment:
add a <remarks> section describing the behaviour of same-key checking and in case of new key.